tests/drum/test_units.py

```diff
@@ -629,250 +742,553 @@ def sparse_df(self):
+                [Values.NUM],
+                "iris_binary",
                 "Species",
-                "10k_diabetes.csv",
+                "ten_k_diabetes",
                 "readmitted",
             ),
             (
-                "NOT_IN",
-                "TXT",
-                "iris_binary_training.csv",
+                Conditions.NOT_IN,
+                [Values.TXT],
+                "iris_binary",
                 "SepalLengthCm",
-                "10k_diabetes.csv",
+                "ten_k_diabetes",
                 "readmitted",
             ),
             (
-                "NOT_EQUALS",
-                "CAT",
-                "iris_binary_training.csv",
+                Conditions.NOT_EQUALS,
+                [Values.CAT],
+                "iris_binary",
                 "Species",
-                "lending_club_reduced.csv",
+                "lending_club",
                 "is_bad",
             ),
             (
-                "EQUALS",
-                "IMG",
-                "cats_dogs_small_training.csv",
+                Conditions.EQUALS,
+                [Values.IMG],
+                "cats_and_dogs",
                 "class",
-                "10k_diabetes.csv",
+                "ten_k_diabetes",
                 "readmitted",
             ),
         ],
+        ids=lambda x: str([str(el) for el in x]) if isinstance(x, list) else str(x),
     )
     def test_data_types(
-        self, condition, value, passing_dataset, passing_target, failing_dataset, failing_target
+        self,
+        condition,
+        value,
+        passing_dataset,
+        passing_target,
+        failing_dataset,
+        failing_target,
+        request,
     ):
-        validator = DataTypes(condition, value)
-        good_data = pd.read_csv(os.path.join(self.tests_data_path, passing_dataset))
+        yaml_str = input_requirements_yaml(Fields.DATA_TYPES, condition, value)
+        schema_dict = self.yaml_str_to_schema_dict(yaml_str)
+        validator = SchemaValidator(schema_dict)
+
+        # good_data = pd.read_csv(os.path.join(self.tests_data_path, passing_dataset))
+        good_data = request.getfixturevalue(passing_dataset)
         good_data.drop(passing_target, inplace=True, axis=1)
-        assert len(validator.validate(good_data)) == 0
-        bad_data = pd.read_csv(os.path.join(self.tests_data_path, failing_dataset))
+        assert validator.validate_inputs(good_data)
+
+        # bad_data = pd.read_csv(os.path.join(self.tests_data_path, failing_dataset))
+        bad_data = request.getfixturevalue(failing_dataset)
         bad_data.drop(failing_target, inplace=True, axis=1)
-        assert len(validator.validate(bad_data)) > 0
+        with pytest.raises(DrumSchemaValidationException):
+            validator.validate_inputs(bad_data)
+
+    def test_data_types_raises_error_if_all_type_in_in_are_not_present(self, iris_binary):
+
+        # TODO is this really correct behavior? if the dataset is missing any one
```

Comment:
@scottp-ml is this correct?  this seems like an odd way to do `IN`

Reply:
IN means that those data types must be present.

Reply:
as noted in a slack conversation, this is mirroring the DR code, even though this is actually `EQUALS`.  

So it's doing what it should. :(